### PR TITLE
Fix/124403 asset loading

### PR DESCRIPTION
**Pull request by cliffordp**
https://central.tri.be/issues/124403

### CODE REVIEW

File: src/Tribe/Main.php [hunk 1]
```diff
@@ -575,11 +573,17 @@ public function hooks() {
 			add_filter( 'tribe_event_import_rsvp_column_names', [ Tribe__Tickets__CSV_Importer__Column_Names::instance(), 'filter_rsvp_column_names' ] );
 		}
 
-		// Load our assets
-		add_action( 'tribe_tickets_plugin_loaded', tribe_callback( 'tickets.assets', 'enqueue_scripts' ) );
-		add_action( 'tribe_tickets_plugin_loaded', tribe_callback( 'tickets.assets', 'admin_enqueue_scripts' ) );
-		add_action( 'admin_enqueue_scripts', tribe_callback( 'tickets.assets', 'enqueue_editor_scripts' ) );
-		add_filter( 'tribe_asset_data_add_object_tribe_l10n_datatables', tribe_callback( 'tickets.assets', 'add_data_strings' ) );
+		/**
```
Comment (sc0ttkclark) [inline src/Tribe/Main.php]: Did we need this change?

File: src/template-tags/tickets.php [hunk 2]
```diff
@@ -1324,6 +1325,60 @@ function tribe_tickets_is_event_page( $post = null ) {
...
+
+		// Front-end archive
+		if ( is_post_type_archive( $post_types ) ) {
+			return true;
+		}
+
+		/**
+		 * Whether or not we are in tickets-enabled context, such as determining if we should load plugin assets.
+		 *
+		 * @param bool           $result
```
Comment (sc0ttkclark) [inline src/template-tags/tickets.php]:
```suggestion
		 * @since TBD
		 *
		 * @param bool           $result
```

File: src/Tribe/Main.php (hunk 1 above)
Comment (cliffordp) [reply src/Tribe/Main.php]: _need?_ no. helps IDE though.

Comment (sc0ttkclark) [reply src/Tribe/Main.php]: IDE maybe but the difference between tribe() and tribe_callback() is one sets up the object, one is a callback that sets up the object and calls the method. I could be wrong and maybe the assets object gets setup ahead of time so this is negligible but the replacement of tribe_callback() is something we'll want to weigh carefully.

Comment (cliffordp) [reply src/Tribe/Main.php]: @sc0ttkclark 
Isn't that what `$assets = tribe( 'tickets.assets' );` is doing (just below this line in GitHub's inline commenting)?

**Are you saying these 2 are slightly different?**

Before:
Screen Shot 2019-12-08 at 12 31 57 PM


After:
image

Comment (sc0ttkclark) [reply src/Tribe/Main.php]: Yes, tribe_callback() returns a temporary callback reference, it does not setup tribe() for that associated container until the callback function is ultimately called via PHP when the action itself runs.

Comment (cliffordp) [reply src/Tribe/Main.php]: tyvm for clarifying

**Pull request merged**